batch/batch/driver/compute_manager.py

```diff
@@ -0,0 +1,80 @@
+import abc
+from typing import Any, Dict, Tuple
+
+from ..worker_config import WorkerConfig
+
+from .zone_monitor import BaseZoneMonitor
+from .activity_monitor import BaseActivityMonitor
+from .disk_monitor import BaseDiskMonitor
+
+import logging
+log = logging.getLogger('compute_manager')
+
+
+class InstanceDoesNotExist(Exception):
+    pass
+
+
+class InstanceState:
+    CREATING = 'Creating'
+    RUNNING = 'Running'
+    TERMINATED = 'Terminated'
+
+    def __init__(self, state, full_spec, last_start_timestamp=None):
+        self.state = state
+        self.full_spec = full_spec
+        self.last_start_timestamp = last_start_timestamp
+
+
+class BaseComputeManager(abc.ABC):
+    activity_monitor: BaseActivityMonitor
+    disk_monitor: BaseDiskMonitor
+    zone_monitor: BaseZoneMonitor
+    credentials: Any
+
+    def __init__(self, app, machine_name_prefix):
+        self.app = app
+        self.machine_name_prefix = machine_name_prefix
+
+    @abc.abstractmethod
+    async def delete_instance(self, instance):
+        pass
+
+    @abc.abstractmethod
+    async def get_instance(self, instance) -> InstanceState:
+        pass
+
+    @abc.abstractmethod
+    async def create_instance(self, *args, **kwargs):
+        pass
```

Comment:
Hmm, I see we use the Instance class for interacting with GCP *and* tracking our InstanceCollection statistics. This feels like an entanglement of two distinct ideas (i.e. a BatchWorker in our Collection versus a VM in the cloud).

Hrm. I really feel like we should not entangle these ideas because it ruins this easy to understand interface.

Assuming we don't separate these ideas, then I think we'll need this interface:

```
def prepare(self, zone, machine_name, machine_type, ...) -> Instance
def create(self, instance):
def get(self, instance) -> InstanceState
def delete(self, instance) -> None

```